lnwallet/commitment.go

```diff
@@ -613,6 +613,42 @@ func CommitScriptAnchors(chanType channeldb.ChannelType,
 	return localAnchor, remoteAnchor, nil
 }
 
+// CommitAuxLeaves stores two potential auxiliary leaves for the remote and
+// local output that made be used to argument the final tapscript trees of the
+// commitment transaction.
+type CommitAuxLeaves struct {
+	// LocalAuxLeaf is the local party's auxiliary leaf.
+	LocalAuxLeaf input.AuxTapLeaf
+
+	// RemoteAuxLeaf is the remote party's auxiliary leaf.
+	RemoteAuxLeaf input.AuxTapLeaf
+}
+
+// ForRemoteCommit returns the local+remote aux leaves from the PoV of the
+// remote party's commitment.
+func (c *CommitAuxLeaves) ForRemoteCommit() CommitAuxLeaves {
+	return CommitAuxLeaves{
+		LocalAuxLeaf:  c.RemoteAuxLeaf,
+		RemoteAuxLeaf: c.LocalAuxLeaf,
+	}
+}
+
+// AuxLeafStore is used to optionally fetch auxiliary tapscript leaves for the
+// commitment transaction given an opaque blob. This is also used to implement
+// a state transition function for the blobs to allow them to be refreshed with
+// each state.
+//
+// TODO(roasbeef): move into pkg for custom chans?
+type AuxLeafStore interface {
+	// FetchLeaves attempts to fetch the auxiliary leaves for the
+	// commitment.
+	FetchLeaves(chanPoint wire.OutPoint,
```

Comment:
I'm not sure I understand this method signature. Why would you pass in the `auxBlob`? Isn't that what contains the aux leaves already? So it would be more of a parse/extract than a fetch? Or what am I misunderstanding?

Reply:
From lines 636-637
> // AuxLeafStore is used to optionally fetch auxiliary tapscript leaves for the
// commitment transaction **given an opaque blob**.

So seems like it's expected to provide the blob. I do see your argument tho, perhaps worth renaming something here

Reply:
Yeh I can probably rename things a bit, it's more of a parse/extract, although the implementor _could_ also store that information, but my latest thinking is that it's derived on the fly. 

I think the "shape" of these interfaces will likely change a bit as we start on the concrete implementations elsewhere. 

The layer of indirection here is that `lnd` doesn't know exactly what's in the blob, but given the blob and the commitment it can obtain the aux tap leaves.